### PR TITLE
API tests for tables, columns and rows

### PR DESCRIPTION
**Pull request by Pedro-Revez-Silva**
## Description
Added the scenario to test operations on Tables, Columns and Rows. Also made some small refactors to previous code.

### CODE REVIEW

File: qa-core/src/tests/internal-api/applications/create.spec.ts [hunk 1]
```diff
@@ -185,11 +171,72 @@ describe("Internal API - /applications endpoints", () => {
...
-    const [response, app] = await config.applications.create(generateApp())
-    expect(response).toHaveStatusCode(200)
-    expect(app.appId).toBeDefined()
+    const app = await config.applications.create(generateApp())
 
     const [deleteResponse] = await config.applications.delete(<string>app.appId)
     expect(deleteResponse).toHaveStatusCode(200)
   })
+
+  it("Operations on Tables", async () => {
```
Comment (Rory-Powell) [inline qa-core/src/tests/internal-api/applications/create.spec.ts:180]: Doesn't seem quote right for this to live under `applications`. Given all the assertions here are specific to a table, this should probably live in the table suite

File: qa-core/src/config/internal-api/fixtures/table.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { Row } from "@budibase/types"
+
+type Table = {
```
Comment (Rory-Powell) [inline qa-core/src/config/internal-api/fixtures/table.ts:3]: We can import the existing definition for a table from budibase. If there is data missing from that definition we should add to the existing one so we aren't repeating ourselves

File: qa-core/src/config/internal-api/TestConfiguration/tables.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        return [response, json]
+    }
+
+    async update(body: any): Promise<[Response, Table]> {
+        const response = await this.api.put(`/tables`, { body })
+        const json = await response.json()
+        return [response, json]
+    }
+
+    async getRows(id: string): Promise<[Response, Row[]]> {
```
Comment (Rory-Powell) [inline qa-core/src/config/internal-api/TestConfiguration/tables.ts:46]: These row operations should probably live in a `RowsApi`

File: qa-core/src/config/internal-api/TestConfiguration/tables.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        return [response, json]
+    }
+
+    async getTableById(id: string): Promise<[Response, Table]> {
+        const response = await this.api.get(`/tables/${id}`)
+        const json = await response.json()
+        return [response, json]
+    }
+
+    async create(body: any): Promise<[Response, Table]> {
```
Comment (Rory-Powell) [inline qa-core/src/config/internal-api/TestConfiguration/tables.ts:28]: should probably call this `save`, given it's used for the create and the update

File: qa-core/src/config/internal-api/fixtures/table.ts (hunk 2 above)
Comment (Pedro-Revez-Silva) [reply qa-core/src/config/internal-api/fixtures/table.ts:3]: Done

File: qa-core/src/config/internal-api/TestConfiguration/tables.ts (hunk 3 above)
Comment (Pedro-Revez-Silva) [reply qa-core/src/config/internal-api/TestConfiguration/tables.ts:46]: Done

File: qa-core/src/config/internal-api/TestConfiguration/tables.ts (hunk 4 above)
Comment (Pedro-Revez-Silva) [reply qa-core/src/config/internal-api/TestConfiguration/tables.ts]: Done

File: qa-core/src/config/internal-api/TestConfiguration/rows.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import InternalAPIClient from "./InternalAPIClient"
+
+export default class RowsApi {
+    api: InternalAPIClient
+
+    constructor(apiClient: InternalAPIClient) {
+        this.api = apiClient
+    }
+
+    async getAll(id: string): Promise<[Response, Row[]]> {
```
Comment (Rory-Powell) [inline qa-core/src/config/internal-api/TestConfiguration/rows.ts:12]: Might make sense to make this explicit with `tableId`

File: packages/types/src/documents/app/table.ts [hunk 6]
```diff
@@ -49,4 +49,10 @@ export interface Table extends Document {
   sourceId?: string
   relatedFormula?: string[]
   constrained?: string[]
+  _id?: string
+  _rev?: string
```
Comment (Rory-Powell) [inline packages/types/src/documents/app/table.ts:53]: We already get these base fields `_id`, `_rev`, `createdAt` and `_updatedAt` from the base `DocumentType`

Comment (Pedro-Revez-Silva) [reply packages/types/src/documents/app/table.ts:53]: Solved

File: qa-core/src/config/internal-api/TestConfiguration/rows.ts (hunk 5 above)
Comment (Pedro-Revez-Silva) [reply qa-core/src/config/internal-api/TestConfiguration/rows.ts:12]: Artifact from when this method was part of tables class. Thank you for pointing that out

**Review by Mitch-Budibase — APPROVED** [review]
LGTM - Nice work Pedro!

**Review by shogunpurple — APPROVED** [review]
LGTM! Nice work Pedro

**Pull request merged**